src/document/advanced.rs

```diff
@@ -1,3 +1,7 @@
+//! Advanced document types
```

Comment:
I don't think I like the word "types" here. Maybe "Advanced document operations"? I think it would also be good to put some text here describing the tradeoff between managed and unmanaged encryption.

Reply:
I chose "types" because this isn't where you find the operations, that's `DocumentAdvancedOps`. This file is just 2 structs and a trait. So I'm good with changing the wording, but I'd rather leave "operations" for the trait itself.

Reply:
I guess I kind of see your point from the prospective of the doc site since the *Result types get re-exported from the `internal` module. The main thing in the module is a trait which doesn't fit my definition of a "type". Maybe "Advanced Document functionality" or "Advanced Document API"?

Reply:
I went with "Advanced document API".
Similar change done for the regular document API.